src/main/java/Crystal.java

```diff
@@ -0,0 +1,80 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
+public class Crystal {
+    public static void main(String[] args) {
```

Comment:
Perhaps the program flow in "main" could be simplified with the use of helper functions?